uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -726,135 +754,116 @@ define([
+     * the date in the request instance.
      */
-    function _convertDateFormatFromServer (date) {
-      return moment(date, sharedSettings.serverDateFormat).toDate();
-    }
-
-    /**
-     * Converts given date to server format
-     *
-     * @param {Date} date
-     * @return {String} date converted to server format
-     */
-    function _convertDateToServerFormat (date) {
-      return moment(date).format(sharedSettings.serverDateFormat);
-    }
-
-    /**
-     * Initializes user's calendar (work patterns)
-     *
-     * @return {Promise}
-     */
-    function _loadCalendar () {
-      return Calendar.get(vm.request.contact_id, vm.period.start_date, vm.period.end_date)
-        .then(function (usersCalendar) {
-          vm.calendar = usersCalendar;
-        });
-    }
-
-    /**
-     * Sets dates (and times in case of "hours" absence type) for vm.request from UI
-     */
-    function _setDates () {
+    function setRequestDates () {
       var options = vm.uiOptions;
       var request = vm.request;
-      var times;
+      var times = options.times;
 
-      request.from_date = options.fromDate ? vm._convertDateToServerFormat(options.fromDate) : null;
-      request.to_date = options.toDate ? vm._convertDateToServerFormat(options.toDate) : null;
+      request.from_date = options.fromDate ? convertDateToServerFormat(options.fromDate) : null;
+      request.to_date = options.toDate ? convertDateToServerFormat(options.toDate) : null;
 
       if (isCalculationUnit('hours') && !isLeaveType('toil')) {
-        times = options.times;
         request.from_date = request.from_date && times.from.time ? request.from_date + ' ' + times.from.time : null;
         request.to_date = request.to_date && times.to.time ? request.to_date + ' ' + times.to.time : null;
-        request.from_date_amount = !isNaN(+times.from.amount) ? times.from.amount : null;
-        request.to_date_amount = !isNaN(+times.to.amount) ? times.to.amount : null;
       }
 
       if (!options.multipleDays && options.fromDate) {
-        options.toDate = options.fromDate;
         request.to_date = request.from_date;
+        request.to_date_type = request.from_date_type;
       }
     }
 
     /**
-     * Sets dates and types for vm.request from UI
+     * Sets dates (with time in case of "hours" Absence Type) from UI to vm.request
+     *
+     * @param  {String} dateType from|to
+     * @return {Promise}
      */
-    function _setDateAndTypes () {
-      vm._setDates();
+    function setDate (dateType) {
+      setMinMaxDatesToUI();
+      checkToDate();
+      resetUIDayTypesTimeAndDeductions(dateType);
 
-      if (!vm.uiOptions.multipleDays && vm.uiOptions.fromDate) {
-        vm.request.to_date_type = vm.request.from_date_type;
-      }
+      vm.uiOptions.times[dateType].loading = true;
+      vm.loading[dateType + 'DayTypes'] = true;
+
+      return $q.resolve()
+        .then(vm.onDateChange)
```

Comment:
There is no need in having noop here since noop simply returns "undefined". It will return undefined anyway if vm.onDateChangeExtended does not exist.